File: data_demo/src/main/scala/com/scala/study/http/HttpUtils.scala
package com.scala.study.http

//import com.mashape.unirest.http.Unirest
//
//class HttpUtils(url:String) {
//
//  def doGet(): Unit ={
//    val jsonResponse = Unirest.get(url).asJson()
//    val jsonNode = jsonResponse.getBody
//    val obj = jsonNode.getObject
//    println(obj)
//  }
//}
//
//object HttpUtils{
//  def main(args: Array[String]): Unit = {
//    val httpUtils = new HttpUtils("http://10.33.15.106:6666/metrics")
//    httpUtils.doGet()
//  }
//}

File: flink_study/src/main/java/com/zyh/utils/HttpUtils.java
package com.zyh.utils;

import com.google.common.base.Optional;
import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.JsonNode;
import com.mashape.unirest.http.Unirest;
import org.json.JSONObject;

/**
 *  远程接口调用测试
 */
public class HttpUtils {

    public static void main(String[] args) throws Exception{
        HttpResponse<JsonNode> jsonResponse =
                Unirest.get("http://10.33.15.106:6666/metrics")  //请求的url
                //.queryString("name","zhangyunhao")     //传递参数
                        .asJson();                                 //转换为json

        JsonNode jsonNode = jsonResponse.getBody();
        JSONObject obj = jsonNode.getObject();
        Optional<JSONObject> op = Optional.of(obj);

        if(op.isPresent()) {
            System.out.println(obj);
        }
//        System.out.println(obj.getString("name"));
//        System.out.println(obj.getInt("code"));
    }
}
